lib/cpp/src/thrift/concurrency/ThreadManager.cpp

```diff
@@ -421,7 +416,7 @@ void ThreadManager::Impl::removeWorker(size_t value) {
   {
     Synchronized s(workerMonitor_);
 
-    while (workerCount_ != workerMaxCount_) {
+    while (workerCount_ > goalCount) {
```

Comment:
Old line 424 was accessing ``workerMaxCount_`` outside of a lock so its behavior was unpredictable when multiple threads called ``removeWorker()`` simultaneously.  Saving the value ti ``goalCount`` removes this unpredictability and therefore is an improvement.  Also see THRIFT-3233 for another fix submitted by someone else.